ProboCI/probo
# Feature/lol/build status errors

As part of the effort to return meaningful errors and a redirection mechanism from the proxy, we need a more structured error message that so that the proxy can watch for certain known conditions.

This PR adds error messages with error codes to the response when a build cannot be returned. Documentation can be found in the readme.

### To Test
 - Create a few builds as usual through your local probo instance.
 - Shutdown the container manager
 - Use lev to update the container manager data and change a build's status to "reaped" by setting reaped: true in the JSON. Note that you need an updated version of lev. (see https://github.com/0x00A/lev/commit/15b1fe31252bd5cf8eac1b68db16c9227a7d664a)
 - Use lev to set another build to in process by changing the "container" property to false.
 - Start the container manager and use curl to request builds.
    - ```curl -X POST -H "Authorization: Bearer devToken" http://localhost:3020/container/proxy?build=[BUILD_ID]```
 - Check to make sure you get the proper responses.
 - Run the unit tests and ensure the new container manager tests pass.
